Refactor BSD build support to be extendable and add NetBSD

vazub opened the pull request:
Allow building `wren_cli` for both FreeBSD and NetBSD, and allow trivial future extension for other BSDs, if needed.

Unfortunately, Premake bundles all separate BSDs under a single `system:bsd` definition, making it impossible to set up build conditionals purely with `--os=` option, and therefore adding additional option was required to circumvent this problem. 

Moreover, it made sense to define dedicated `gmake.<*bsd>` folder location for each system's project files, to have clear separation of concerns, and since BSD `make` is not usable in our case, folder prefix was deliberately changed to `gmake.` to provide additional hint for those building the project from source, regarding what `make` variant they are required to use.

Commit:
Refactor BSD build support to be extendable and add NetBSD

Commit:
Fix some broken formatting for premake5.lua file

Not sure how and why it got borked, but editing in Gihub Web interface seems to have solved it.

vazub commented on projects/premake/premake5.lua:
```diff
@@ -1,164 +1,182 @@
+  defaultplatform "64bit"
+  location ("../" .. _ACTION)
+
+  newoption {
```
Added this

ruby0x1 commented:
Thanks @vazub looks like a reasonable approach. I'll maybe rearrange a tiny bit but otherwise will test/review properly soon.

Commit:
got my eyes blurred, fix stupid mistake

Commit:
One more build clarification

Don't assume that people would be building on Free/NetBSD host system, be explicit.

vazub commented:
Looks like there is no interest in this. Closing.

ruby0x1 commented:
The issue isn't really interest, mostly time, and the burden of testing and maintaining those platforms for each release is a consideration. 

I think this is a reasonable approach as mentioned, though there's a file that has it's entire contents marked as a diff which we should probably correct before this can be merged either way.

ruby0x1 commented on projects/premake/premake5.lua:
```diff
@@ -1,164 +1,182 @@
-workspace "wren-cli"
```
Let's fix up the line endings or whatever has caused this entire file to be a diff :)

Commit:
Merge branch 'wren-lang:main' into netbsd

Commit:
whitespace fixes

Commit:
even more whitespace fixes

vazub commented:
@ruby0x1 kind request to address my question in the resolved section above, because I am not sure how to proceed.
